usrp_drivers/n200/n200_driver.cpp

```diff
@@ -382,245 +398,253 @@ std::string random_string( size_t length )
-          << std::chrono::duration_cast<std::chrono::microseconds>(send_end - send_begin).count()
-          << "us" << std::endl;
-  }
-}
 
-void control(zmq::context_t* driver_c) {
-  std::cout << "Enter control thread\n";
+    data_socket.send(samples_metadata_size_message);
 
-  std::cout << "Creating and connecting to thread socket in control\n";
-/*    zmq::socket_t packet_socket(*thread_c, ZMQ_PAIR); // 1
-  packet_socket.connect("inproc://threads"); // 2  */
+    auto send_end = std::chrono::steady_clock::now();
+    time_diff = std::chrono::duration_cast<std::chrono::microseconds>
+                                                  (send_end - send_begin).count();
+    DEBUG_MSG("RECEIVE package and send timing: " << time_diff << "us");
+  }
+}
 
+/**
+ * @brief      Runs in a seperate thread to act as an interface for the ingress and egress data.
+ *
+ * @param[in]  driver_c        The driver ZMQ context.
+ */
+void control(zmq::context_t &driver_c) {
+  DEBUG_MSG("Enter control thread");
 
-  zmq::socket_t packet_socket(*driver_c, ZMQ_PUB);
-  packet_socket.bind("inproc://threads");
+  DEBUG_MSG("Creating and connecting to thread socket in control");
 
-  std::cout << "Creating and binding control socket\n";
-  //zmq::context_t context(2);
-  zmq::socket_t radarctrl_socket(*driver_c, ZMQ_PAIR);
-  radarctrl_socket.bind("ipc:///tmp/feeds/0");
+  zmq::socket_t driver_packet_pub_socket(driver_c, ZMQ_PUB);
+  ERR_CHK_ZMQ(driver_packet_pub_socket.bind("inproc://threads"))
 
+  DEBUG_MSG("Creating and binding control socket");
+  zmq::socket_t radarctrl_socket(driver_c, ZMQ_PAIR);
+  ERR_CHK_ZMQ(radarctrl_socket.bind("ipc:///tmp/feeds/0"))
   zmq::message_t request;
 
-  zmq::socket_t data_socket(*driver_c, ZMQ_PAIR);
-  data_socket.bind("inproc://data");
+  zmq::socket_t data_socket(driver_c, ZMQ_PAIR);
+  ERR_CHK_ZMQ(data_socket.bind("inproc://data"))
 
-  zmq::socket_t ack_socket(*driver_c, ZMQ_PAIR);
-  ack_socket.bind("inproc://ack");
+  zmq::socket_t ack_socket(driver_c, ZMQ_PAIR);
+  ERR_CHK_ZMQ(ack_socket.bind("inproc://ack"))
 
-  zmq::socket_t computation_socket(*driver_c, ZMQ_PAIR);
-  computation_socket.connect("ipc:///tmp/feeds/1");
+  zmq::socket_t rx_dsp_socket(driver_c, ZMQ_PAIR);
+  ERR_CHK_ZMQ(rx_dsp_socket.connect("ipc:///tmp/feeds/1"))
 
 
 
-  //Allows for set up of SUB/PUB socket
+  //Sleep to handle "slow joiner" problem
+  //http://zguide.zeromq.org/php:all#Getting-the-Message-Out
   sleep(1);
 
   while (1) {
-    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
     radarctrl_socket.recv(&request);
-    std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
 
-    begin = std::chrono::steady_clock::now();
-    driverpacket::DriverPacket dp;
+    auto begin_deserialize = std::chrono::steady_clock::now();
+    driverpacket::DriverPacket driver_packet;
     std::string msg_str(static_cast<char*>(request.data()), request.size());
```

Comment:
Can you rename msg_str to request_str to reflect that it is a string coming from the request variable? This is used in each of the three threads

Reply:
done this.